vlsi/Makefile

```diff
@@ -54,23 +55,20 @@ endif
 #########################################################################################
 # general rules
 #########################################################################################
-ALL_RTL = $(TOP_FILE) $(TOP_SMEMS_FILE)
-extra_v_includes = $(build_dir)/EICG_wrapper.v
 ifneq ($(CUSTOM_VLOG), )
 	VLSI_RTL = $(CUSTOM_VLOG)
-	VLSI_BB = /dev/null
+	VLSI_BB =
 else
-	VLSI_RTL = $(ALL_RTL) $(extra_v_includes)
-	VLSI_BB = $(sim_top_blackboxes)
+	VLSI_RTL = $(shell cat $(TOP_MODS_FILELIST)) $(TOP_SMEMS_FILE) 
+	# TODO: have MFC split top & harness blackboxes
+	VLSI_BB = $(build_dir)/EICG_wrapper.v $(shell rev $(BB_MODS_FILELIST) | sed -E 's/cc(.*)/c\1/g' | uniq -s 1 | sed '/c\./d' | rev)
```

Comment:
- I think having this oneliner will probably result in confusion down the line. Can you add a comment on what it's doing.
- Technically, `VLSI_RTL` could contain BB's if the BB filename matches the name of the module. I'm not sure if that matters. I guess why is there a distinction between the BB's and VLSI_RTL?
- Side note: IIRC most synth. tools will do dead code elimination and thus will eliminate unused modules. So do we really need to split up the BB's between top/model? I understand from a code cleanliness perspective but the solution seems fairly heavy-handed when a downstream tool can deal with this well.

Reply:
1. Sure, can clarify what it's doing.
2. So that I'm clear what you mean - blackboxes where the resource name is equivalent to the module's name get put in the `.top.f` file, and those where the names are different go into `.bb.f`? I can totally merge `VLSI_RTL` and `VLSI_BB` - the latter was just leftover from the previous Makefile.
3. Theoretically you are correct. However, the issue here is that synthesis tools don't support the DPI calls that are present in the harness' blackboxes (SimJTAG, etc.), erroring out as a result. They generally only like purely behavioral + structural Verilog.

Reply:
> 2. So that I'm clear what you mean - blackboxes where the resource name is equivalent to the module's name get put in the `.top.f` file, 

This is true.

> and those where the names are different go into `.bb.f`?

The `bb.f` file holds all BB files for both the top/harness (will include the BB's put in the `top.f` file).

> I can totally merge `VLSI_RTL` and `VLSI_BB` - the latter was just leftover from the previous Makefile.

Combining `top.f` and `bb.f` (ofc removing the non-verilog files) and removing duplicate BB's files will give you all the Verilog files for the TOP and some Verilog BB files that reside in the MODEL.

> 3. Theoretically that should be true. However, the issue here is that synthesis tools don't support the DPI calls that are present in the harness' blackboxes (SimJTAG, etc.), erroring out as a result. They generally only like purely behavioral + structural Verilog.

Agreed. You at minimum need to remove the non-Verilog files.

Reply:
currently working on 3

Reply:
@harrisonliew  @abejgonzalez  can you guys check if this is the behavior that we want for separating top.bb.f & model.bb.f?

Reply:
Thanks for fixing 3! Updated the Makefile to combine into `VLSI_RTL` and use the new split script.